### PR TITLE
Adding namespace label related tests

### PR DESCRIPTION
**Pull request by vpinisetti-px**
**What this PR does / why we need it**:
This is to automate Namespace label related tests

**Which issue(s) this PR fixes** (optional)
Closes #
https://portworx.atlassian.net/browse/PA-770
https://portworx.atlassian.net/browse/PA-817
https://portworx.atlassian.net/browse/PA-818

**Special notes for your reviewer**:
All tests in single run:
https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/1598/

Individual Tests:
https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/1528/console

https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/1526/console

https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/1584/console

### CODE REVIEW

File: tests/backup/backup_namespace_labelled_test.go [hunk 1]
```diff
@@ -430,3 +430,772 @@ var _ = Describe("{BackupScheduleForOldAndNewNS}", func() {
...
+			err := DeleteRestore(restoreName, orgID, ctx)
+			dash.VerifySafely(err, nil, fmt.Sprintf("Verifying the deletion of the restore named [%s]", restoreName))
+		}
+		log.InfoD("Deleting labels from namespaces - %v", bkpNamespaces)
+		err = DeleteLabelsFromMultipleNamespaces(nsLabels, bkpNamespaces)
+		dash.VerifySafely(err, nil, fmt.Sprintf("Deleting labels [%v] from namespaces [%v]", nsLabels, bkpNamespaces))
+		opts := make(map[string]bool)
+		opts[SkipClusterScopedObjects] = true
+		log.InfoD("Deleting deployed namespaces - %v", bkpNamespaces)
+		ValidateAndDestroy(scheduledAppContexts, opts)
```
Comment (sn-px) [inline tests/backup/backup_namespace_labelled_test.go:620]: use DestroyApps in cleanup

File: tests/backup/backup_namespace_labelled_test.go [hunk 2]
```diff
@@ -430,3 +430,772 @@ var _ = Describe("{BackupScheduleForOldAndNewNS}", func() {
...
+		dash.VerifySafely(err, nil, fmt.Sprintf("Deleting backup schedule policies %s ", []string{periodicSchPolicyName}))
+		for _, restoreName := range restoreNames {
+			err := DeleteRestore(restoreName, orgID, ctx)
+			dash.VerifySafely(err, nil, fmt.Sprintf("Verifying the deletion of the restore named [%s]", restoreName))
+		}
+		bkpNamespaces = append(bkpNamespaces, newNamespaces...)
+		opts := make(map[string]bool)
+		opts[SkipClusterScopedObjects] = true
+		log.InfoD("Deleting deployed namespaces - %v", bkpNamespaces)
+		ValidateAndDestroy(scheduledAppContexts, opts)
```
Comment (sn-px) [inline tests/backup/backup_namespace_labelled_test.go:847]: use DestroyApps in claeanup

File: tests/backup/backup_namespace_labelled_test.go [hunk 3]
```diff
@@ -430,3 +430,772 @@ var _ = Describe("{BackupScheduleForOldAndNewNS}", func() {
...
+		err = Inst().Backup.DeleteBackupSchedulePolicy(orgID, []string{periodicSchPolicyName})
+		dash.VerifySafely(err, nil, fmt.Sprintf("Deleting backup schedule policies %s ", []string{periodicSchPolicyName}))
+		for _, restoreName := range restoreNames {
+			err := DeleteRestore(restoreName, orgID, ctx)
+			dash.VerifySafely(err, nil, fmt.Sprintf("Verifying the deletion of the restore named [%s]", restoreName))
+		}
+		opts := make(map[string]bool)
+		opts[SkipClusterScopedObjects] = true
+		log.InfoD("Deleting deployed namespaces - %v", bkpNamespaces)
+		ValidateAndDestroy(scheduledAppContexts, opts)
```
Comment (sn-px) [inline tests/backup/backup_namespace_labelled_test.go:1198]: use DestroyApps in cleanup

**Review by sn-px — APPROVED** [review]

File: tests/backup/backup_namespace_labelled_test.go (hunk 2 above)
Comment (kshithijiyer-px) [reply tests/backup/backup_namespace_labelled_test.go:850]: Please use DestroyApps here

File: tests/backup/backup_namespace_labelled_test.go (hunk 1 above)
Comment (kshithijiyer-px) [reply tests/backup/backup_namespace_labelled_test.go:620]: Please use DestroyApps here

File: tests/backup/backup_namespace_labelled_test.go (hunk 3 above)
Comment (vpinisetti-px) [reply tests/backup/backup_namespace_labelled_test.go]: Done

File: tests/backup/backup_namespace_labelled_test.go (hunk 2 above)
Comment (vpinisetti-px) [reply tests/backup/backup_namespace_labelled_test.go]: Done

File: tests/backup/backup_namespace_labelled_test.go (hunk 1 above)
Comment (vpinisetti-px) [reply tests/backup/backup_namespace_labelled_test.go]: Done

File: tests/backup/backup_namespace_labelled_test.go [hunk 4]
```diff
@@ -430,3 +430,772 @@ var _ = Describe("{BackupScheduleForOldAndNewNS}", func() {
...
+				restoredNameSpace := fmt.Sprintf("%s-%v", RestoreNamePrefix, time.Now().Unix())
+				scheduleAllNSRestoreMapping[namespace] = restoredNameSpace
+			}
+			customRestoreName := fmt.Sprintf("%s-%v", schBackupAllNS, time.Now().Unix())
+			err = CreateRestore(customRestoreName, incrementalRestoreAllNS, scheduleAllNSRestoreMapping, SourceClusterName, orgID, ctx, nil)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verification of restoring scheduled backups for all application [%s] in custom namespace [%v]", customRestoreName, scheduleAllNSRestoreMapping))
+			restoreNames = append(restoreNames, customRestoreName)
+		})
+	})
+	JustAfterEach(func() {
```
[the remark this answers is not in the document]
Comment (vpinisetti-px) [reply tests/backup/backup_namespace_labelled_test.go:1185]: Done

**Review by kshithijiyer-px — APPROVED** [review]
LGTM

**Comment by mkoppal-px** [comment]
Can we run all 3 tests together and make sure nothing is breaking?

**Pull request merged**